src/Cookie/CookieJar.php

```diff
@@ -86,6 +86,21 @@ public static function shouldPersist(
         return false;
     }
 
+    /**
+     * Finds and returns the cookie based on the name
+     *
+     * @param string $name cookie name to search for
+     * @return SetCookie|null cookie that was found or null if not found
+     */
+    public function getCookieByName($name)
+    {
+        foreach($this->cookies as $cookie) {
+            if(strcasecmp($cookie->getName(), $name) === 0) {
```

Comment:
You could have a `SetCookie` with name = null and that would fail on HHVM. ([3v4l](https://3v4l.org/oLT0P)). That is an edge case but I believe we should cover it.